### PR TITLE
feature: expose the 'Last-Modified' response header as ngx.header["Last-Modified"].

### PR DESCRIPTION
**Pull request by zhuizhuhaomeng**
I hereby granted the copyright of the changes in this pull request
to the authors of this lua-nginx-module project.

the original PR is https://github.com/openresty/lua-nginx-module/pull/1384.
the ngx.header was rewrite via ffi. So need to rewrite this feature.

### CODE REVIEW

**Review by xiaocang — CHANGES_REQUESTED** [review]

**Review by xiaocang — APPROVED** [review]

File: src/ngx_http_lua_headers.c [hunk 1]
```diff
@@ -1134,6 +1135,22 @@ ngx_http_lua_ffi_get_resp_header(ngx_http_request_t *r,
...
+            p = ngx_palloc(r->pool, sizeof("Mon, 28 Sep 1970 06:00:00 GMT"));
+            if (p == NULL) {
+                *errmsg = "no memory";
+                return NGX_ERROR;
+            }
+
+            values[0].data = p;
+            values[0].len = ngx_http_time(p, last_modified) - p;
+            return 1;
+        }
```
Comment (doujiang24) [inline src/ngx_http_lua_headers.c:1150]: maybe we can return 0 here when the `last_modified < 0`? just for better performance.

File: src/ngx_http_lua_headers.c [hunk 2]
```diff
@@ -1134,6 +1135,22 @@ ngx_http_lua_ffi_get_resp_header(ngx_http_request_t *r,
 
         break;
 
+    case 13:
+        last_modified = r->headers_out.last_modified_time;
```
Comment (doujiang24) [inline src/ngx_http_lua_headers.c:1139]: we should check if the `key_buf` is `Last-Modified-time`?

File: t/016-resp-header.t [hunk 3]
```diff
@@ -2141,3 +2141,63 @@ hi
...
+--- request
+GET /a.txt
+--- raw_response_headers_like chomp
+Age: \d\r\n
+--- no_error_log
+[error]
+
+
+
+=== TEST 96: Expose the 'Last-Modified' response header as ngx.header["Last-Modified"]
```
Comment (doujiang24) [inline t/016-resp-header.t:2176]: duplicated test name, better use a unique one.
Do you want to test the header is not existing?

File: src/ngx_http_lua_headers.c [hunk 4]
```diff
@@ -1134,6 +1135,29 @@ ngx_http_lua_ffi_get_resp_header(ngx_http_request_t *r,
 
         break;
 
+    case 13:
+        if (ngx_strncasecmp(key_buf, (u_char *) "Last-Modified", 13) == 0) {
+
```
Comment (doujiang24) [inline src/ngx_http_lua_headers.c:1140]: style: useless blank line.

**Review by doujiang24 — APPROVED** [review]

**Pull request merged**